### PR TITLE
[Finishes #106352136] String contains

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Add String.prototype.contains to applab

Add some tests for this and our other string functions

### CODE REVIEW

File: apps/src/applab/dropletConfig.js [hunk 1]
```diff
@@ -121,11 +120,12 @@ module.exports.blocks = [
...
-  {func: 'toUpperCase', blockPrefix: 'str.toUpperCase', category: 'Variables', 'modeOptionName': '*.toUpperCase', type: 'value' },
-  {func: 'toLowerCase', blockPrefix: 'str.toLowerCase', category: 'Variables', 'modeOptionName': '*.toLowerCase', type: 'value' },
+  {func: 'substring', blockPrefix: 'str.substring', category: 'Variables', paletteParams: ['start','end'], params: ["6", "11"], modeOptionName: '*.substring', type: 'value' },
+  {func: 'indexOf', blockPrefix: 'str.indexOf', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.indexOf', type: 'value' },
+  {func: 'contains', blockPrefix: 'str.contains', category: 'Variables', paletteParams: ['searchValue'], params: ['"World"'], modeOptionName: '*.contains', type: 'value' },
+  {func: 'length', block: 'str.length', category: 'Variables', modeOptionName: '*.length' },
+  {func: 'toUpperCase', blockPrefix: 'str.toUpperCase', category: 'Variables', modeOptionName: '*.toUpperCase', type: 'value' },
+  {func: 'toLowerCase', blockPrefix: 'str.toLowerCase', category: 'Variables', modeOptionName: '*.toLowerCase', type: 'value' },
```
Comment (Bjvanminnen) [inline apps/src/applab/dropletConfig.js:128]: Note, most of these changes are getting rid of quotes around modeOptionName

File: apps/src/codegen.js [hunk 2]
```diff
@@ -247,10 +245,8 @@ exports.makeNativeMemberFunction = function (opts) {
         nativeArgs[i] = exports.marshalInterpreterToNative(opts.interpreter, arguments[i]);
       }
       var nativeRetVal = opts.nativeFunc.apply(opts.nativeParentObj, nativeArgs);
-      return exports.marshalNativeToInterpreter(opts.interpreter,
-                                                nativeRetVal,
-                                                null,
-                                                opts.maxDepth);
+      return exports.marshalNativeToInterpreter(opts.interpreter, nativeRetVal,
+        null, opts.maxDepth);
```
Comment (Bjvanminnen) [inline apps/src/codegen.js:249]: No functional change - just changing indentation.

**Comment by davidsbailey** [comment]
This seems like a pretty big departure from our strategy up to this point, which was to only expose methods on classes which actually have that method in Javascript (ES5). Can you give me some background on this decision?

**Comment by Bjvanminnen** [comment]
I agree, and expressed my apprehensiveness about monkey patching String.prototype.

Basically, the ed team thinks that teaching indexOf() !== -1 right away is more confusing than they would like. I suggested possibly doing something like contains(string, substring), but that's also weird since we do have a bunch of other methods on String.prototype that we teach.

After discussing some we Sarah, we decided that the benefits (easier for students) outweighed the fact that we're now teaching stuff that won't work in native JS. If you feel strongly about this, you can try to convince her otherwise :)

**Comment by Bjvanminnen** [comment]
Will update to use String.prototype.includes instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

**Comment by Bjvanminnen** [comment]
@davidsbailey could you take a look now that we're using includes?

**Comment by davidsbailey** [comment]
LGTM

**Pull request merged**